unit flake: TestControllerSync index out of range

From https://console.cloud.google.com/m/cloudstorage/b/kubernetes-jenkins/o/pr-logs/pull/26710/kubernetes-pull-test-unit-integration/28875/artifacts/junit_v1%2Cautoscaling-v1%2Cbatch-v1%2Cbatch-v2alpha1%2Cextensions-v1beta1%2Capps-v1alpha1%2Cfederation-v1alpha1%2Cpolicy-v1alpha1%2Crbac.authorization.k8s.io-v1alpha1_20160602-071507.stdout:

```
--- FAIL: TestControllerSync (0.05s)
panic: runtime error: index out of range [recovered]
	panic: runtime error: index out of range

goroutine 124 [running]:
panic(0x1c74920, 0xc820012070)
	/usr/local/go/src/runtime/panic.go:481 +0x3ff
testing.tRunner.func1(0xc8201be120)
	/usr/local/go/src/testing/testing.go:467 +0x22a
panic(0x1c74920, 0xc820012070)
	/usr/local/go/src/runtime/panic.go:443 +0x521
k8s.io/kubernetes/pkg/controller/persistentvolume.TestControllerSync.func2(0xc820327450, 0xc820348230, 0x1f5bb80, 0x12, 0xc8201c6420, 0x1, 0x1, 0xc8201c6428, 0x1, 0x1, ...)
	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/controller/persistentvolume/controller_test.go:68 +0x13a
k8s.io/kubernetes/pkg/controller/persistentvolume.TestControllerSync(0xc8201be120)
	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/controller/persistentvolume/controller_test.go:122 +0x19c1
testing.tRunner(0xc8201be120, 0x2aa88b0)
	/usr/local/go/src/testing/testing.go:473 +0xdd
created by testing.RunTests
	/usr/local/go/src/testing/testing.go:582 +0xae3
FAIL	k8s.io/kubernetes/pkg/controller/persistentvolume	1.499s
```

--- a/pkg/controller/persistentvolume/controller_test.go
+++ b/pkg/controller/persistentvolume/controller_test.go
@@ -111,8 +111,12 @@ func TestControllerSync(t *testing.T) {
 		defer ctrl.Stop()
 		go ctrl.Run()
 
-		// Wait for the controller to pass initial sync.
-		for !ctrl.volumeController.HasSynced() || !ctrl.claimController.HasSynced() {
+		// Wait for the controller to pass initial sync and fill its caches.
+		for !ctrl.volumeController.HasSynced() ||
+			!ctrl.claimController.HasSynced() ||
+			len(ctrl.claims.ListKeys()) < len(test.initialClaims) ||
+			len(ctrl.volumes.store.ListKeys()) < len(test.initialVolumes) {
+
 			time.Sleep(10 * time.Millisecond)
 		}
 		glog.V(4).Infof("controller synced, starting test")

